Cabal/src/Distribution/PackageDescription/Check/Warning.hs

```diff
@@ -0,0 +1,1017 @@
+    ++ "internal executable: "
+    ++ commaSep (map prettyShow depInternalExecWithImpossibleVersion)
+    ++ ". This version range does not include the current package, and must "
+    ++ "be removed as the current package's executable will always be used."
+ppExplanation (MissingInternalExe depInternalExeWithImpossibleVersion) =
+  "The package depends on a missing internal executable: "
+    ++ commaSep (map prettyShow depInternalExeWithImpossibleVersion)
+ppExplanation NONELicense = "The 'license' field is missing or is NONE."
+ppExplanation NoLicense = "The 'license' field is missing."
+ppExplanation AllRightsReservedLicense =
+  "The 'license' is AllRightsReserved. Is that really what you want?"
+ppExplanation (LicenseMessParse lic) =
+  "Unfortunately the license "
+    ++ quote (prettyShow lic)
+    ++ " messes up the parser in earlier Cabal versions so you need to "
+    ++ "specify 'cabal-version: >= 1.4'. Alternatively if you require "
+    ++ "compatibility with earlier Cabal versions then use 'OtherLicense'."
+ppExplanation (UnrecognisedLicense l) =
+  quote ("license: " ++ l)
+    ++ " is not a recognised license. The "
+    ++ "known licenses are: "
+    ++ commaSep (map prettyShow knownLicenses)
+ppExplanation UncommonBSD4 =
+  "Using 'license: BSD4' is almost always a misunderstanding. 'BSD4' "
+    ++ "refers to the old 4-clause BSD license with the advertising "
+    ++ "clause. 'BSD3' refers the new 3-clause BSD license."
+ppExplanation (UnknownLicenseVersion lic known) =
+  "'license: "
+    ++ prettyShow lic
+    ++ "' is not a known "
+    ++ "version of that license. The known versions are "
+    ++ commaSep (map prettyShow known)
+    ++ ". If this is not a mistake and you think it should be a known "
+    ++ "version then please file a ticket."
+ppExplanation NoLicenseFile = "A 'license-file' is not specified."
+ppExplanation (UnrecognisedSourceRepo kind) =
+  quote kind
+    ++ " is not a recognised kind of source-repository. "
+    ++ "The repo kind is usually 'head' or 'this'"
+ppExplanation MissingType =
+  "The source-repository 'type' is a required field."
+ppExplanation MissingLocation =
+  "The source-repository 'location' is a required field."
+ppExplanation MissingModule =
+  "For a CVS source-repository, the 'module' is a required field."
+ppExplanation MissingTag =
+  "For the 'this' kind of source-repository, the 'tag' is a required "
+    ++ "field. It should specify the tag corresponding to this version "
+    ++ "or release of the package."
+ppExplanation SubdirRelPath =
+  "The 'subdir' field of a source-repository must be a relative path."
+ppExplanation (SubdirGoodRelPath err) =
+  "The 'subdir' field of a source-repository is not a good relative path: "
+    ++ show err
+ppExplanation (OptFasm fieldName) =
+  "'"
+    ++ fieldName
+    ++ ": -fasm' is unnecessary and will not work on CPU "
+    ++ "architectures other than x86, x86-64, ppc or sparc."
+ppExplanation (OptViaC fieldName) =
```

Comment:
In fact, from an early GHC 7.x release on, `-fvia-C` is a no-op unless you are using an unregisterised compiler, in which case it is the default.

I should probably dig out where this is generated and flag it there, but this is where I noticed it.

Reply:
https://downloads.haskell.org/ghc/9.8.1/docs/users_guide/codegens.html#c-code-generator-fvia-c fwiw. For a number of versions after this change `-fvia-C` was converted to `-asm` internally with a warning, but now it is an error except in an unregisterised compiler.

Reply:
Nicely spotted, I will erase this.

Reply:
After a bit of digging:
- It does not seem to error *as now*, [just a deprecation warning](https://gitlab.haskell.org/ghc/ghc/-/blob/master/compiler/GHC/Driver/Session.hs#L1809-1814)
- still this is caught by GHC and cabal warning adds nothing, so I am going to axe it anyway.

Reply:
Interesting. The GHC User's Guide says otherwise, although `-fvia-C` is confusingly listed under warning options in the flag summary.